build/rpmfc.c

```diff
@@ -444,17 +444,12 @@ int rpmfcExec(ARGV_const_t av, StringBuf sb_stdin, StringBuf * sb_stdoutp,
     }
     ec = getOutputFrom(xav, buf_stdin, buf_stdin_len, sb,
 		       failnonzero, buildRoot, dup);
-    if (ec) {
-	sb = freeStringBuf(sb);
-    }
```

Comment:
This has consequences for callers who expect the former behavior. At least it introduces a memory leak to rpmfcApplyExternal() AFAICS.

Reply:
I can restore this with just adding `goto exit;` if that is what you would prefer.

Reply:
I didn't say I want it restored, I just said it has consequences that need to be dealt with.

The question is whether we actually want to change the behavior. I looked at this very thing when reviewing the dynamic buildrequires patches and while wanting the output might seem obvious, we actually have it already because it's duplicated to actual stdout descriptor in the cases where we care about the output. Or that's how it looks to me, but do you have a concrete case where something important is not shown because of this?

Reply:
to be honest, I just felt weird that it was freed when failnonzero.. I did not have any specific case why I wanted it to be different.

SO I suppose if we change behavior, we should change behavior of return code because now it is just 0 - ok, 1 - fail. while in some cases you want to get actual return code. 

So I propose to merge it without behavior change so we don't have to fix anything else on the way. And when we decide to change behavior, then we will look at it again.

WDYT?

Reply:
What I think is that this whole move to rpmfcExec() has started increasingly  feeling like a mistake. :grimacing: But it's a bit late to revert, that'd be a huge mess too.

I agree lets just leave it the way it is for now.